### DISCUSSION TITLE
Cannot seem to expose application via Beta images

### DISCUSSION THREAD

**Discussion by magweter** [serversideup/docker-php#328] [Q&A] [answered]
Hi all,

The serversideup/php:beta-8.2-fpm-nginx image worked perfectly for me! unfortunately just up until a few days ago. Now I cannot seem to get my application to expose through Nginx. I just constantly get a connection refused. When I use the serversideup/php:8.2-fpm-nginx image, my (laravel) application does expose and works like it should. I'm using SSL off and port 8080 (mapped from container port 80) in my browser.

I require the installation of some php-extensions which drives me to use the beta image, as the serversideup/php:8.2-fpm-nginx image does not seem to be able to run install-php-extensions.
![image](https://github.com/serversideup/docker-php/assets/27769920/446b63c4-f1b0-428c-a31b-df6c7da0a2c2)

I'm really trying to stick to install-php-extensions to escape the apt-get and pecl install hassle.

My dockerfile:
```Dockerfile
# Set our base image
FROM serversideup/php:beta-8.2-fpm-nginx as base

ENV SSL_MODE=off
ENV PHP_OPCACHE_ENABLE=1

# Switch to root before installing our PHP extensions
USER root

RUN install-php-extensions zlib
RUN install-php-extensions sqlsrv
RUN install-php-extensions pdo_sqlsrv

FROM base as production

COPY --chown=www-data:www-data . .

# Switch back to the unprivileged www-data user
USER www-data
```

Building succeeds and everything seems ok:
<img width="1058" alt="image" src="https://github.com/serversideup/docker-php/assets/27769920/96ffe008-c54a-457e-9af5-09e32daa02a6">

**Comment by jaydrogers** [comment]
Thanks for the Dockerfile.

I ran a build with your file and it seemed to work just fine:

```
docker build -t localhost:latest .
```
![CleanShot 2024-04-26 at 16 28 36@2x](https://github.com/serversideup/docker-php/assets/3174134/ba7e90df-ad1a-46b0-beb4-c3f92b1f2d6a)

### Your error
![image](https://github.com/serversideup/docker-php/assets/3174134/1e19868c-813f-4704-80c9-f50eef9c8f0f)

The screenshot you have looks like it's failing on build.

Is this when you're using Docker Compose with a build command?

If so, be sure to add `--build --force-recreate` to make sure it's not a weird cache:

```bash
docker compose up --build --force-recreate
```

**Reply by magweter** [reply]
Thanks for your answer. The build for beta works (I only get the error using the non-beta image). The problem was actually suddenly not being able to expose my application through the beta image, which made me try switching to the non-beta image.

**Comment by magweter** [comment]
I have found the problem. Took me some time though.. I've been looking into the container layers for answers. The image I built a week ago using php:8.2-fpm-nginx as base image has EXPOSE 80,443 as build steps;

<img width="754" alt="image" src="https://github.com/serversideup/docker-php/assets/27769920/e8e046bd-2b98-4efe-93a5-8b3100a492ae">


Now this seems to have changed, as the most recent image uses EXPOSE 8080,8443;
<img width="745" alt="image" src="https://github.com/serversideup/docker-php/assets/27769920/deef041b-758e-42bf-acd1-92fb91bf0d4d">

Thus changing my mapping from 8080:80 to 8080:8080 solved the issue.

I get that the beta image is subject to change. I've been using it as the non-beta 8.2 fpm nginx image does not seem to support install-php-extensions, is that correct? In that case I would love for the image to be released or for the beta image to have versioning to prevent surprises ;-)

**Reply by jaydrogers** [reply, marked as answer]
The "non-beta" is still "v2".

The "beta" is v3, which is an entire rewrite. I suggest reading this guide: https://serversideup.net/open-source/docker-php/docs/guide/migrating-from-v2-to-v3

If you're starting with a new project, start with the "beta" as that will be released to stable shortly.

If you have an existing project, be sure to lock into `v2.2.1` as it's shown in the docs until you have time to refactor with the new v3 images 👍
